src/server/log.js

```diff
@@ -43,80 +45,99 @@ class Logger {
 		// Only log if given a high enough priority level.
-		if (level.priority > this.level.priority) {
-			return;
-		}
-		let messageToLog = `[${level.name}@${new Date(Date.now()).toISOString()}] ${message}\n`;
-
-		// Add a stacktrace to the message if one was provided
-		if (error !== null) {
-			if (error.stack) {
-				messageToLog += `Stacktrace: \n${error.stack}\n`;
-			} else {
-				// It's possible someone passed in an error that isn't actually an Error object
-				// because javascript lets you throw anything. In that case, the error won't have
-				// a stack.
-
-				// So we just try to convert whatever we got to a string and include it. It's better
-				// than nothing.
-				messageToLog += `An error was included, but it was not an Error object:\n${error}`;
+		if (level.priority <= this.level.priority && !skipMail) {
+			let messageToLog = `[${level.name}@${new Date(Date.now()).toISOString()}] ${message}\n`;
+
+			// Add a stacktrace to the message if one was provided
+			if (error !== null) {
+				if (error.stack) {
+					messageToLog += `Stacktrace: \n${error.stack}\n`;
+				} else {
+					// It's possible someone passed in an error that isn't actually an Error object
+					// because javascript lets you throw anything. In that case, the error won't have
+					// a stack.
+
+					// So we just try to convert whatever we got to a string and include it. It's better
+					// than nothing.
+					messageToLog += `An error was included, but it was not an Error object:\n${error}`;
+				}
 			}
-		}
-		if (this.logToConsole) {
-			if (level.priority >= LogLevel.WARN.priority) {
-				// tslint:disable-next-line no-console
-				console.error(messageToLog);
-			} else {
-				// tslint:disable-next-line no-console
-				console.log(messageToLog);
+			if (this.logToConsole) {
+				if (level.priority >= LogLevel.WARN.priority) {
+					// tslint:disable-next-line no-console
+					console.error(messageToLog);
+				} else {
+					// tslint:disable-next-line no-console
+					console.log(messageToLog);
+				}
+			}
+			if (this.logToFile) {
+				fs.appendFile(logFile, messageToLog, err => {
+					if (err) {
+						console.error(`Failed to write to log file: ${err}`); // tslint:disable-line no-console
+					}
+				});
 			}
 		}
-		if (this.logToFile) {
-			fs.appendFile(logFile, messageToLog, err => {
-				if (err) {
-					console.error(`Failed to write to log file: ${err}`); // tslint:disable-line no-console
+
+		// Only send an e-mail if given a high enough priority level.
+		if (level.priority <= this.emailLevel.priority) {
+			let messageToMail = `At ${new Date(Date.now()).toISOString()}, an ${level.name} event occurred.\n`;
+			messageToMail += `${message}\n`;
+			// Add a stacktrace to the message if one was provided
+			if (error !== null) {
+				if (error.stack) {
+					messageToMail += `Stacktrace: \n${error.stack}\n`;
+				} else {
+					messageToMail += `An error was included, but it was not an Error object:\n${error}`;
```

Comment:
Is this text really necessary to email?

Reply:
I am not sure. Do you want to include this in the email @huss ?